libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -104,566 +145,647 @@ TEST_SETUP( TEST_IOT_UART )
 }
 /*-----------------------------------------------------------*/
-
-/**
- * @brief Test function to test the asynchronous read with UART by doing a
- * hardware loopback. The Tx and Rx pin on the vendor board are shorted with a
- * connector and the test is run by transmitting some bytes on write async and checking
- * if same characters are read aynchronously.
- */
-TEST( TEST_IOT_UART, AFQP_IotUARTWriteReadAsyncWithCallback )
-{
-    uint8_t cpBuffer[ testBufferSize ] = { 0 };
-    uint8_t cpBufferRead[ testBufferSize ] = { 0 };
-
-    strncpy( ( char * ) cpBuffer, ( char * ) testIotUART_READ_BUFFER_STRING, testIotUART_READ_BUFFER_LENGTH );
-    IotUARTHandle_t xOpen;
-    int32_t lRead, lWrite, lClose;
-    BaseType_t xCallbackReturn;
-    uint8_t ucPort = ustestIotUartPort;
-
-    xOpen = iot_uart_open( ucPort );
-    TEST_ASSERT_NOT_EQUAL( NULL, xOpen );
-
-    if( TEST_PROTECT() )
-    {
-        iot_uart_set_callback( xOpen, prvReadWriteCallback, NULL );
-
-        lWrite = iot_uart_write_async( xOpen, cpBuffer, testIotUART_READ_BUFFER_LENGTH );
-        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lWrite );
-
-        xCallbackReturn = xSemaphoreTake( ( SemaphoreHandle_t ) &xtestIotUARTCompleted, testIotUART_DEFAULT_SEMPAHORE_DELAY );
-        TEST_ASSERT_EQUAL( pdTRUE, xCallbackReturn );
-
-        lRead = iot_uart_read_async( xOpen, cpBufferRead, testIotUART_READ_BUFFER_LENGTH );
-        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lRead );
-
-        xCallbackReturn = xSemaphoreTake( ( SemaphoreHandle_t ) &xtestIotUARTCompleted, testIotUART_DEFAULT_SEMPAHORE_DELAY );
-        TEST_ASSERT_EQUAL( pdTRUE, xCallbackReturn );
-
-        TEST_ASSERT_EQUAL( 0, strncmp( ( char * ) cpBuffer, ( char * ) cpBufferRead, testIotUART_READ_BUFFER_LENGTH ) );
-    }
-
-    lClose = iot_uart_close( xOpen );
-    TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lClose );
-}
 /*-----------------------------------------------------------*/
 
 /**
  * hardware loopback. The Tx and Rx pin on the vendor board are shorted with a
  * connector and the test is run by transmitting some bytes on write sync and checking
  * if same characters are read synchronously.
  */
-TEST( TEST_IOT_UART, AFQP_IotUARTWriteReadSync )
+TEST( TEST_IOT_UART, AFQP_AssistedIotUARTReadWriteSync )
 {
-    uint8_t cpBuffer[ testBufferSize ] = { 0 };
-    uint8_t cpBufferRead[ testBufferSize ] = { 0 };
+    uint8_t cpBuffer[ testIotUARTBUFFERSIZE ] = { 0 };
+    uint8_t cpBufferRead[ testIotUARTBUFFERSIZE ] = { 0 };
 
-    strncpy( ( char * ) cpBuffer, ( char * ) testIotUART_WRITE_BUFFER_STRING, testIotUART_WRITE_BUFFER_LENGTH );
-    IotUARTHandle_t xOpen;
+    strncpy( ( char * ) cpBuffer, ( char * ) testIotUART_BUFFER_STRING, testIotUART_BUFFER_LENGTH );
+    IotUARTHandle_t xUartHandle;
     int32_t lRead, lWrite, lClose;
-    uint8_t ucPort = ustestIotUartPort;
 
-    xOpen = iot_uart_open( ucPort );
-    TEST_ASSERT_NOT_EQUAL( NULL, xOpen );
-    if( TEST_PROTECT() )
```

Comment:
I would like to keep the TEST_PROTECT() lines around the tests to make sure that the UART stays in a good state if one test fails

Reply:
Added TEST_PROTECT()